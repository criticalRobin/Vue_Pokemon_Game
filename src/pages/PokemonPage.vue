<template>
  <h1 v-if="!pokemon">Espere por favor...</h1>

  <div v-else>
    <h1>Quién es este Pokémon?</h1>
    <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon" />
    <PokemonOptions :pokemonsArr="pokemonsArr" />
  </div>
</template>

<script>
import PokemonPicture from "@/components/PokemonPicture.vue";
import PokemonOptions from "@/components/PokemonOptions.vue";
import getPokemonOptions from "@/helpers/getPokemonOptions";

export default {
  components: {
    PokemonPicture,
    PokemonOptions,
  },
  data() {
    return {
      pokemonsArr: [],
      pokemon: null,
      showPokemon: false,
    };
  },
  methods: {
    async getPokemons() {
      this.pokemonsArr = await getPokemonOptions();
      const rndInt = Math.floor(Math.random() * 4);
      this.pokemon = this.pokemonsArr[rndInt];
    },
  },
  mounted() {
    this.getPokemons();
  },
};
</script>
