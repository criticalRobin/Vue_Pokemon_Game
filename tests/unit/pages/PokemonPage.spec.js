import { shallowMount, mount } from "@vue/test-utils";
import PokemonPage from "@/pages/PokemonPage.vue";
import { pokemonsArr } from "../mocks/pokemons.mock";

describe("pruebas unitarias del componente PokemonPage", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test("debe llamar al getPokemons al montar", () => {
    const getPokemonsSpy = jest.spyOn(PokemonPage.methods, "getPokemons");
    shallowMount(PokemonPage);

    expect(getPokemonsSpy).toHaveBeenCalled();
  });

  test("debe hacer match con el snapchot cuando cargan los pokemon", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonsArr,
          pokemon: pokemonsArr[0],
          showPokemon: false,
          showAnswer: false,
          message: "",
          lives: [1, 2, 3],
          gameOver: false,
        };
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
