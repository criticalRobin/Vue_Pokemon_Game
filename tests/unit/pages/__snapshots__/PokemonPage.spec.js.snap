// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas unitarias del componente PokemonPage debe hacer match con el snapchot cuando cargan los pokemon 1`] = `
<div>
  <h1>Quién es este Pokémon?</h1>
  <lives-stub lives="1,2,3"></lives-stub>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemonsarr="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
</div>
<!--v-if-->
`;
